mllib/src/main/scala/org/apache/spark/ml/feature/StopWordsRemover.scala

```diff
@@ -98,13 +67,16 @@ class StopWordsRemover(override val uid: String)
 
   /**
    * the stop words set to be filtered out
-   * Default: [[StopWords.English]]
+   * Default: [[StopWords.languageMap("english")]]
    * @group param
    */
   val stopWords: StringArrayParam = new StringArrayParam(this, "stopWords", "stop words")
 
   /** @group setParam */
-  def setStopWords(value: Array[String]): this.type = set(stopWords, value)
+  def setStopWords(value: Array[String]): this.type = {
+    set(stopWords, value)
+    set(language, "unknown")
```

Comment:
We shouldn't set other parameters in the setters since their are other ways to overwrite param values. Let's keep the default language to `english` and if `stopWords` is set, ignore the language setting. This logic should happen in `transform`.